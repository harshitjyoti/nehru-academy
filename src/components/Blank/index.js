import React from 'react';
import { Result } from 'antd';
const Blank = () => <>
  <Result title="Please execute a search" />
</>
export default Blank;