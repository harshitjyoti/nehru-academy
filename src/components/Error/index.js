import React from 'react';
import { Result, Button } from 'antd';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  render() {
    const { hasError } = this.state;
    return <>{hasError ?
      <Result
        status="warning"
        title="There are some problems with your operation."
        extra={
          <Button type="primary" key="console" onClick={() => this.props.history.push('/')}>
            Go To Login
                        </Button>}>
      </Result> :
      this.props.children
    }
    </>
  }
}

export default ErrorBoundary;