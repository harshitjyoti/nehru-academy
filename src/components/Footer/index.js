import React from 'react';
import { Layout, Pagination } from 'antd';

const Footer = ({ count, onSearch }) => <Layout.Footer style={{
    textAlign: 'center',
    position: 'fixed',
    width: '100%',
    bottom: 0
}}>
    <Pagination defaultCurrent={1} total={count} onChange={(page) => onSearch(`page=${page}`)} showSizeChanger={false} />
</Layout.Footer>



export default Footer;