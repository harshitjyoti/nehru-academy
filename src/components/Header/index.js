import React from 'react';
import { Layout, Input, Popover, Anchor, Avatar } from 'antd';
let timer;
const Header = ({ onSearch, isProcessing, user, logout }) => {
  const searchValue = (event) => {
    const { value } = event.target
    clearTimeout(timer)
    timer = setTimeout(() => {
      onSearch(`search=${value}`);
    }, 500)
  }
  return (<Layout.Header style={{ position: 'fixed', zIndex: 1, width: '100%', padding: 0 }}>
    <div style={{ textAlign: "center" }}>
      <Input.Search onChange={searchValue} style={{ width: '50%' }} enterButton={false} size='middle' maxLength='20' placeholder="Search for Planets here !" loading={isProcessing} />
      <Popover content={<Anchor onClick={logout}><Anchor.Link title='Logout' /></Anchor>} title="You Can:">
        <Avatar style={{
          float: 'right',
          margin: 13
        }} size="large">
          {user.charAt(0)}
        </Avatar>
      </Popover>
    </div>
  </Layout.Header>)
}

export default Header;