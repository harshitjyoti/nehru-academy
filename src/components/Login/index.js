import React from 'react';
import { Button, Input, Layout, Card, message } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loginAction, setUser } from '../../store';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: null,
      password: null
    };
  }

  error = () => {
    message.error('Login Failed ! Wrong Credentials !');
    this.props.setUser()

  };

  setUser = (event) => {
    const { id, value } = event.target;
    this.setState({ [id]: value })
  }

  login = () => {
    const { user, password } = this.state;
    this.props.loginAction({ user, password })
  }

  componentDidUpdate() {
    const { isUserFetched, userDetails, history, hasError } = this.props;
    isUserFetched && userDetails && history.push('/planets');
    isUserFetched && !userDetails && this.error()
    hasError && history.push('/error');
  }


  render() {
    const { isProcessing } = this.props;
    const { user, password } = this.state;
    return (
      <Layout.Content style={{
        textAlign: "center",
        top: '50%',
        position: 'absolute',
        left: '50%',
        msTransform: 'translateY(-50%) translateX(-50%)',
        transform: 'translateY(-50%) translateX(-50%)',
        width: '40%'
      }}>
        <div style={{
          textAlign: "center"
        }}>
          <Card title={<><UserOutlined /> Login to Star-Wars</>}>
            <Input disabled={isProcessing} size="large" placeholder="User Name" id='user' prefix={<UserOutlined />} onChange={this.setUser} /><hr />
            <Input.Password disabled={isProcessing} size="large" placeholder="Password" id='password' onChange={this.setUser} /><br /><br />
            <Button disabled={!user || !password} type="primary" onClick={this.login} loading={isProcessing}>
              Login
                </Button>
          </Card>
        </div>
      </Layout.Content>
    );
  }
}

const mapStateToProps = ({ user: { isProcessing, userDetails, hasError, isUserFetched } }) => {
  return {
    isProcessing,
    userDetails,
    hasError,
    isUserFetched
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ loginAction, setUser }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);