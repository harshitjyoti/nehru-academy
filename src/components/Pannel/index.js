import React from 'react';
import { Collapse, Badge } from 'antd';
import { getColor } from '../../helpers';
import { icons } from '../../constants';
Math.floor(Math.random() * 10)
const getExtra = (population, color) => <Badge style={{ backgroundColor: color }} count={parseInt(population).toLocaleString()} />

const getHeader = (name, color) => <>
{icons[Math.floor(Math.random() * 10)]}
<span style={{ color, fontWeight: 'bold', textDecoration: "underline" }} >{name}</span>
</>
const Pannel = ({ planets }) => {
  return planets.map((item, index) => {
    const color = getColor(item.population)
    return <Collapse key={index}>
      <Collapse.Panel header={getHeader(item.name, color)} key="1" extra={getExtra(item.population, color)}>
        {Object.keys(item).map((i, index) => (
          <span key={index}><b>{i.replace(/_/g, ' ').toUpperCase()}</b> : {item[i]}<hr /></span>
        ))}
      </Collapse.Panel>
    </Collapse>
  }
  )
}

export default Pannel;