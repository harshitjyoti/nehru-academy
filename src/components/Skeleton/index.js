import React from 'react';
import { Skeleton as Skull } from 'antd';

const Skeleton = () => <>
    <Skull active />
    <Skull active />
    <Skull active />
    <Skull active />
</>

export default Skeleton;