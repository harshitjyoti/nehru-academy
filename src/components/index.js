export { default as Blank } from './Blank';
export { default as Footer } from './Footer';
export { default as Header } from './Header';
export { default as Skeleton } from './Skeleton';
export { default as Pannel } from './Pannel';
export { default as Login } from './Login';
export { default as ErrorBoundary } from './Error';
export { default as ErrorComp } from './Error/comp';