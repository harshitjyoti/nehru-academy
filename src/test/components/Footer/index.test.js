import React from 'react';
import { Footer } from '../../../components';
import renderer from 'react-test-renderer';

describe('Snapshot ', () => {
  beforeAll(() => {  
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  test('SnapShot test', () => {
    const component = renderer.create(
      <Footer count={5} onSearch={()=>console.log("hi")}/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});