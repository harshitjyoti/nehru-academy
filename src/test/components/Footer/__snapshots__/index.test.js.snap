// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot  SnapShot test 1`] = `
<footer
  className="ant-layout-footer"
  style={
    Object {
      "bottom": 0,
      "position": "fixed",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <ul
    className="ant-pagination"
    style={Object {}}
    unselectable="unselectable"
  >
    <li
      aria-disabled={true}
      className="ant-pagination-prev ant-pagination-disabled"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={null}
      title="Previous Page"
    >
      <a
        className="ant-pagination-item-link"
        disabled={true}
      >
        <span
          aria-label="left"
          className="anticon anticon-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </span>
      </a>
    </li>
    <li
      className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex="0"
      title={1}
    >
      <a>
        1
      </a>
    </li>
    <li
      aria-disabled={true}
      className="ant-pagination-next ant-pagination-disabled"
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={null}
      title="Next Page"
    >
      <a
        className="ant-pagination-item-link"
        disabled={true}
      >
        <span
          aria-label="right"
          className="anticon anticon-right"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
            />
          </svg>
        </span>
      </a>
    </li>
  </ul>
</footer>
`;
